### PR TITLE
webpack 2 and config improvments

### PR DESCRIPTION
**Pull request by sokra**
webpack 2 supports HMR for ES6 imports, so the app import can be simplified:

``` diff
-
-function getApp() {
-  return require("./app").default;
-}
+import app from "./app";
```

I also added the `NamedModulesPlugin` which makes update messages look nicer:

``` text
[HMR] Updated modules:
[HMR]  - ./src/middleware/api.js
[HMR]  - ./src/app.js
[HMR] Update applied.
```

instead of

``` text
[HMR] Updated modules:
[HMR]  - 5
[HMR]  - 1
[HMR] Update applied.
```

### CODE REVIEW

**Comment by ericclemmons** [comment]
@sokra Thanks so much for this :) 😊 I was wondering what the new `NamedModulesPlugin` did in Webpack2 and now it makes perfect sense.

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "presets": [
-    "es2015"
+    "es2015-native-modules"
```
Comment (ericclemmons) [inline .babelrc:3]: I've been using https://www.npmjs.com/package/babel-preset-es2015-webpack which looks to be about the same thing (slightly different implementations), but I'm going to use `es2015-native-modules` in my examples going forward, trusting that you know what works better with Webpack than I :)

Comment (kentcdodds) [inline .babelrc:3]: I'm curious about this. Which should I recommend that people use? The `-webpack` or `-native-modules` preset?

File: webpack.config.js [hunk 2]
```diff
@@ -20,7 +20,6 @@ module.exports = {
   module: {
     loaders: [
       {
-        exclude: /node_modules/,
```
Comment (kentcdodds) [inline webpack.config.js:23]: Why was this necessary to remove? Wont this slow down the build considerably?

Comment (wmertens) [inline webpack.config.js:23]: it's the babel default, no?

Comment (ericclemmons) [inline webpack.config.js:23]: Generally, your webpack loaders only get applied to your source files, not dependencies, since they're typically shipped as ES3/5.

Comment (kentcdodds) [inline webpack.config.js:23]: Ummm. Unless I'm mistaken, that's not built into Webpack. Webpack will run loaders on all matching files regardless of whether it's in node_ modules. Babel may default to not transpile node_modules though. That could make sense I guess...

Comment (sokra) [inline webpack.config.js:23]: The `externals` option excludes node_modules from build anyway.

But anyway you should not use `exclude` but `include` instead.

File: .babelrc (hunk 1 above)
Comment (sokra) [inline .babelrc:3]: They seem to be equal.

I looked at the source code and it seems like the `-webpack` is better.
- It removes the plugin from the preset by name, `-native-modules` removes it by index (less error prone)
- It is tested on travis

So we better change it to `-webpack`...

Comment (ericclemmons) [inline .babelrc:3]: I was concerned about the same thing (removing by index).  Glad we're in agreement.

I'll take care of this.

File: webpack.config.js (hunk 2 above)
Comment (kentcdodds) [inline webpack.config.js:23]: Very interesting. Thanks!
